src/OpenApi/src/Comparers/OpenApiSchemaComparer.cs

```diff
@@ -24,6 +24,15 @@ public bool Equals(OpenApiSchema? x, OpenApiSchema? y)
             return true;
         }
 
+        // If a local reference is present, we can't compare the schema directly
+        // and should instead use the schema ID as a type-check to assert if the schemas are
+        // equivalent.
+        if ((x.Reference != null && y.Reference == null)
```

Comment:
Can this be simplified to a XOR?
 (`x.Reference == null ^ y.Reference == null`)?

Reply:
It could although admittedly I find the long-form to be a bit easier to read 😅